include/common/msgType.h

```diff
@@ -633,10 +636,43 @@ struct DDN_FAIL_Q_msg{
 	uint8_t cause;
 };
 #define S11_DDN_FAIL_BUF_SIZE sizeof(struct DDN_FAIL_Q_msg)
+
+struct CB_RESP_Q_msg {
+    msg_type_t msg_type;
+    int ue_idx;
+    uint8_t cause;
+    uint32_t seq_no;
+    bearerCtxList_t bearerCtxList;
+    struct pco pco;
+    struct fteid s11_sgw_c_fteid;
+ };
+ #define S11_CBRESP_BUF_SIZE sizeof(struct CB_RESP_Q_msg)
+ 
+struct DB_RESP_Q_msg {
+    msg_type_t msg_type;
+    int ue_idx;
+    uint32_t seq_no;
+    uint8_t cause;
+    uint8_t linked_bearer_id;
+    bearerCtxList_t bearerCtxList;
+    struct pco pco;
+    struct fteid s11_sgw_c_fteid;
+ };
+#define S11_DBRESP_BUF_SIZE sizeof(struct DB_RESP_Q_msg)
+ 
 /*************************
  * Incoming GTP Messages
  *************************/
+
+typedef struct gtp_incoming_msg_data_t {
+    uint32_t destInstAddr;
+    uint32_t srcInstAddr;
+    msg_type_t msg_type;
+    int ue_idx;
```

Comment:
Even ue_index should not be part of top level header right ?

Reply:
Removed and placed inside the Q_msg as "s11_mme_cp_teid".